### PR TITLE
pallet-scheduler: Ensure we request a preimage

### PR DESCRIPTION
**Pull request by bkchr**
The scheduler was not requesting a preimage. When a preimage is requested, a user can deposit it without paying any fees.

### CODE REVIEW

File: frame/scheduler/src/tests.rs [hunk 1]
```diff
@@ -489,19 +491,23 @@ fn scheduler_handles_periodic_unavailable_preimage() {
...
-		// Unnote the preimage.
+		// Unnote the preimage
 		Preimage::unnote(&hash);
 
 		// Does not ever execute again.
 		run_to_block(100);
 		assert_eq!(logger::log().len(), 1);
-
-		// The preimage is not requested anymore.
-		assert!(!Preimage::is_requested(&hash));
```
Comment (ggwpez) [inline frame/scheduler/src/tests.rs:504]: You can keep this check to assert that a periodic job wont re-request it.

**Review by ggwpez — APPROVED** [review]
Looks like `cancel` was already correctly un-requesting :+1:

**Review by muharem — APPROVED** [review]

**Comment by muharem** [comment]
On service_task now we drop the call assuming its not `Inline`.
We probably need a trait for the call. But this does not concern this PR.

File: frame/scheduler/src/tests.rs [hunk 2]
```diff
@@ -479,8 +480,9 @@ fn scheduler_handles_periodic_unavailable_preimage() {
 		let call = RuntimeCall::Logger(LoggerCall::log { i: 42, weight: (max_weight / 3) * 2 });
 		let hash = <Test as frame_system::Config>::Hashing::hash_of(&call);
 		let len = call.using_encoded(|x| x.len()) as u32;
-		let bound = Preimage::pick(hash, len);
-		assert_ok!(Preimage::note(call.encode().into()));
+		let bound = Bounded::Lookup { hash, len };
```
Comment (melekes) [inline frame/scheduler/src/tests.rs:483]:
```suggestion
        // Important to use here `Bounded::Lookup` to ensure that we request the hash.
		let bound = Bounded::Lookup { hash, len };
```

**Review by melekes — COMMENTED** [review]
lgtm

**Comment by bkchr** [comment]
> On service_task now we drop the call assuming its not `Inline`. We probably need a trait for the call. But this does not concern this PR.

What you mean? We already called drop before on the call.

**Comment by bkchr** [comment]
bot merge

**Comment by bkchr** [comment]
bot merge

**Comment by muharem** [comment]
@bkchr I am talking about places where we do not uses lookup_hash, but inline call is possible
Ex: https://github.com/paritytech/substrate/blob/2ced413e40eed0e294fbc4c0d539bac2376016f6/frame/scheduler/src/lib.rs#L826

**Comment by bkchr** [comment]
Yeah, this was already there and is correct. Internally it handles `Inline` correctly: https://github.com/paritytech/substrate/blob/8047698fe6786ae51e6d0fe811203a6551c82d98/frame/support/src/traits/preimages.rs#L153-L159

**Pull request merged**